core/src/main/java/tachyon/worker/eviction/EvictPartialLFU.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.eviction;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import tachyon.Pair;
+import tachyon.worker.hierarchy.BlockInfo;
+import tachyon.worker.hierarchy.StorageDir;
+
+/**
+ * Used to evict blocks in certain StorageDir by LFU strategy.
```

Comment:
Different from EvictLFU, this PartialLFU first chooses a StorageDir with the max free space. Then it applies LFU algorithm in one StorageDir instead of the whole StorageTier.

It's similar with current EvictPartialLRU. And I think PartialLFU has a better load balance behavior than LFU.